### DISCUSSION TITLE
What the relationship between Vetur and volar?

### DISCUSSION THREAD

**Discussion by lovetingyuan** [vuejs/language-tools#119] [Q&A] [answered]
What the relationship between Vetur and volar?
Which one will be the official plugin?
Does vetur still not support vue3 with typescript(especially for template)?
In other words, volar is better than vetur when using vue3 with typescript, is it right?

**Comment by johnsoncodehk** [comment, marked as answer]
> What the relationship between Vetur and volar?

No relationship.

> Which one will be the official plugin?

Vetur and [VueDX](https://github.com/znck/vue-developer-experience) is official plugins.

> Does vetur still not support vue3 with typescript(especially for template)?

With VueDX it can.

> In other words, volar is better than vetur when using vue3 with typescript, is it right?

I can't answer. Volar has some strong opinions for vue language support. Some typescript behavior may different.

btw you may need to note, Volar will not include ESLint and Prettier, you should install these yourself if you need.

**Reply by lovetingyuan** [reply]
Thanks, much appreciation for your reply
